src/pu-utils.c

```diff
@@ -428,3 +428,22 @@ pu_device_get_partition_path(const gchar *device,
         return NULL;
     }
 }
+
+gchar *
+pu_str_pre_remove(gchar *string,
+                  guint n)
+{
+    guchar *start;
+
+    g_return_val_if_fail(string != NULL, NULL);
+
+    if (n >= strlen(string))
+        n = strlen(string);
```

Comment:
I guess this returns an empyt string? Maybe you can handle this more directly here?

Reply:
I don't see a simpler way to handle this case. Because the function returns *and* modifies `string`, we have to continue here and only adjust `n` to be in range. Directly returning would not result in modification of `string`. The functionality is similar to what the C stdlib functions do while manipulating strings.

Reply:
Yes empty string isn't easier here